### ISSUE TITLE
Big Picture opening windowed

### ISSUE THREAD

**Issue by LuNeder** [ValveSoftware/steam-for-linux#10059]
#### Your system information

* Steam client version (build number or date): 1694466342
* Distribution (e.g. Ubuntu): OpenSUSE Tumbleweed
* Opted into Steam client beta?: yes, but happens on release too since last update (I switched to beta exactly to see if this was fixed)
* Have you checked for system updates?: y
* Steam Logs: [logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/12587421/logs.tar.gz)
* GPU: Nvidia

#### Please describe your issue in as much detail as possible:
The start in windowed mode setting is DISABLED. Enabling and disabling it again makes it go fullscreen, but next time you launch it it'll be windowed again. Big Picture should open in fullscreen (and in the current monitor) when you press the big picture mode button on Steam, but it opens in a small window instead.

#### Steps for reproducing this issue:

1. Open Steam BigPicture mode
2. It will be windowed instead of fullscreen, defeating the whole purpose of big picture

**Comment by kisak-valve** [comment]
Hello @LuNeder, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by codesnippy** [comment]
> #### Your system information
> * Steam client version (build number or date): 1694466342
> * Distribution (e.g. Ubuntu): OpenSUSE Tumbleweed
> * Opted into Steam client beta?: yes, but happens on release too since last update (I switched to beta exactly to see if this was fixed)
> * Have you checked for system updates?: y
> * Steam Logs: [logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/12587421/logs.tar.gz)
> * GPU: Nvidia
> 
> #### Please describe your issue in as much detail as possible:
> The start in windowed mode setting is DISABLED. Enabling and disabling it again makes it go fullscreen, but next time you launch it it'll be windowed again. Big Picture should open in fullscreen (and in the current monitor) when you press the big picture mode button on Steam, but it opens in a small window instead.
> 
> #### Steps for reproducing this issue:
> 1. Open Steam BigPicture mode
> 2. It will be windowed instead of fullscreen, defeating the whole purpose of big picture

I have the exact same thing on Debian Bullseye with NVidia GPU

**Comment by ClocQ** [comment]
I have a same problem on Ubuntu 22.04.03 LTS and N95 Built-in GPU.
It has been occurring for several weeks.

(I rely on the Japanese to English translation of DeepL. I'm sorry if it is difficult to read.)

**Comment by alex-kronish** [comment]
I've been having this issue for about a week or two as well. 

Steam Big Picture simply does not want to honor the setting, I have to toggle the windowed mode on then off for it to be fullscreen every time I open steam.

**Comment by Avsynthe** [comment]
Having the same issue on Debian Bookworm.

Same issue as https://github.com/flathub/com.valvesoftware.Steam/issues/1112

**Comment by v1k0d3n** [comment]
Same issue on Fedora 38 fwiw. Nothing seems to work. Could this be a holdover issue from the default Deck resolution?

**Comment by smcv** [comment]
I have two similarly-configured test machines with multiple OSs and graphical environments installed, so I tried to narrow down in which situations `steam -bigpicture` starts windowed ("bad" below) or starts full-screen (shown as "good" below).

My best guess so far is that this has something to do with the proprietary Nvidia drivers? Is anyone affected by this **not** using an Nvidia GPU and the proprietary drivers?

* AMD Vega, Mesa
    * Arch
        * KDE Plasma, Wayland: good
    * Ubuntu 22.04
        * GNOME, Wayland ("Ubuntu"): good
        * GNOME, Xorg ("Ubuntu on Xorg"): good
* Nvidia proprietary driver
    * Arch
        * KDE Plasma, X11: bad
        * KDE Plasma, Wayland: bad
    * Ubuntu 22.04
        * GNOME, Wayland ("Ubuntu on Wayland"): bad
        * GNOME, Xorg ("Ubuntu"): bad

**Comment by rkfg** [comment]
I use NVIDIA and I'm affected by this issue. Do you see these two lines in the Steam log when running BP on AMD?
```
XRRGetOutputInfo Workaround: initialized with override: 0 real: 0xf0c0e780
XRRGetCrtcInfo Workaround: initialized with override: 0 real: 0xf0c0d050
```
Could be related as these functions belong to xrandr which is used to detect resolution and other display parameters.

**Comment by smcv** [comment]
> Do you see these two lines in the Steam log when running BP on AMD?

Yes I do. The addresses are different, but that's normal: they're different on my Nvidia system too.

> Could be related as these functions belong to xrandr which is used to detect resolution and other display parameters.

Yes, it could be related, but it's not as simple as "that workaround is wrong": it's active on both Mesa (which is working correctly for me) and Nvidia (which isn't).

**Comment by rkfg** [comment]
For those using Awesome WM I made a workaround until it's properly fixed. Add a rule to your `awful.rules.rules` (you should already have some most probably):
```
    { rule = { name = "Steam Big Picture Mode" },
      properties = {
        fullscreen = true
      }
    },
```
I have another good rule for the normal Steam mode and its other windows, I put it above the BP rule so together they look like this:
```
    { rule_any = { class = { "steam", "Steam", "steamwebhelper" } },
      properties = {
        floating = true,
        focus = true,
        border_width = 0,
        tag = "7"
      }
    },
    { rule = { name = "Steam Big Picture Mode" },
      properties = {
        fullscreen = true
      }
    },
```
`tag = "7"` is of course optional, it simply launches Steam on the 7th tag by default. I'm sure something similar exists in other advanced WMs.

**Comment by rkfg** [comment]
The current steam build works fine for me, opens in full screen as expected. Either it was fixed or the Nvidia's driver, I installed the latest beta 545.29.02.

**Comment by buckysrevenge** [comment]
I'll see when the 545 drivers make it to Manjaro stable, but currently my GTX 960 will full screen BPM from startup only in an X11 session, neither it nor my laptop (hybrid Intel/Nvidia RTX2070) will fullscreen BPM at startup in Wayland

**Comment by LuNeder** [comment]
Still opens windowed for me on latest NVidia drivers, both on X11 and Wayland.

In fact it’s now even more bugged after the latest steam (or nvidia?) updates, the window stays black for a while (instead of showing the launch animation) before becoming usable.

**Comment by hadim** [comment]
I am having the same issue on Ubuntu + NVIDIA (wayland and x11).

**Comment by dsiggi** [comment]
Same problem here on Gentoo + NVIDIA (wayland).

**Comment by thndgonz** [comment]
Having the same issue. Weirdly... big picture mode was more stable before as in right now the interface feels like it isn't hardware accelerated/sluggish as heck. 

I thought it was just some weird misconfiguration I did but I'm using Arch Linux/KDE + Nvidia and its been opening in windowed mode for a while now. The interface also used to be snappy... I wouldn't be surprised if nvidia once again screwed something up.

**Comment by JustCryen** [comment]
Alright, so I neglected this because I only use bigscreen from time to time but it started to bother me.

When the new Bigscreen UI first dropped it was super snappy and smooth, literally nothing wrong with it.
Right now I experience a really low performance, stuttering and it won't launch in fullscreen. It will enter fullscreen the moment I switch it in settings but it won't respect this on the next launch.
Basically the same thing everyone here already said. Currently I am on Gnome Wayland but it is also happening on Gnome x11.

The one thing I see in common is that everyone with this problem happen to use Nvidia GPUs. BUT I believe currently my system uses only the integrated intel GPU.
Currently I can't check this behavior on my AMD system but I believe I also experienced something weird on it regarding BigPicture.

[Steam System Info](https://gist.github.com/JustCryen/4d2a17d1c464e7fb09a61ef503922fa2)
Video example **!!!WARNING - gets a bit epileptic at the end!!!**

https://github.com/ValveSoftware/steam-for-linux/assets/41633661/c45c1793-6f0e-4785-a741-a90070d65636


**Os:** Archlinux
**Kernel:** linux-zen 6.7.zen3-1
**GPU:** GeForce GTX 1060 Mobile
**GPU driver:** nvidia-dkms 545.29.06-1

**Comment by JustCryen** [comment]
Honestly… I didn't expect that to happen xP
Took me off guard, I just wanted to try all the possible combinations.

**Comment by autumngray** [comment]
For me, switching to the beta update channel fixed both issues (fullscreen + performance). Unfortunately, now I'm having problems with controller input (#10406)

**Comment by hadim** [comment]
I confirm the beta update channel fixes the fullscreen and performance issues for me (on X11 + Nvidia).

**Comment by buckysrevenge** [comment]
Beta will start full screen for me now as well in Wayland (GTX960 545.29.(something), Manjaro stable, 6.7 kernel), but the menu is still messed up (see image). Otherwise, it is running pretty well.

![PXL_20240120_174848462](https://github.com/ValveSoftware/steam-for-linux/assets/26829479/9523d808-be41-40b2-a144-811172525d5e)

**Comment by LuNeder** [comment]
Can confirm that the issue is fixed in latest beta with latest drivers, so I’m closing this!

(The black big picture startup and invisible menu are still a problem for me, but we should open a separate issue for that if one still doesn’t exist)

Thanks!

**Comment by HamonTeacher** [comment]
I have the same bug on Windows 11 LOL

**Comment by LuNeder** [comment]
> I have the same bug on Windows 11 LOL

Move to Linux then :)

**Comment by HamonTeacher** [comment]
> > I have the same bug on Windows 11 LOL
> 
> Move to Linux then :)

It didn't help you. And your advice is frankly stupid.
